### DISCUSSION TITLE
ValidationElement: ability to pass runtime error messages

### DISCUSSION THREAD

**Discussion by pierrecdn** [zauberzeug/nicegui#3441] [Ideas / Feature Requests]
Instead of applying a static `dict[str, Callable]` validation, complex validation methods may compute their specific output.

Typically, [here](https://github.com/zauberzeug/nicegui/blob/main/nicegui/elements/mixins/validation_element.py#L43) the `check` could also be considered as a `Callable` and return its own message.

For example, in Pydantic context, it could allow to pass general purpose vaidation metadata in addition to very specific validation methods, generally performed after submission.

What do you think?

**Comment by rodja** [comment]
Sounds great. How would the API look? Would you or some else work on a pull request?

**Reply by pierrecdn** [reply]
I gave it a try: https://github.com/zauberzeug/nicegui/pull/3443
